Cython/Compiler/Nodes.py

```diff
@@ -3890,15 +3890,11 @@ def generate_stararg_copy_code(self, code):
                 self.starstar_arg.entry.cname, self.error_value()))
             code.put_gotref(self.starstar_arg.entry.cname, py_object_type)
             code.putln("} else {")
-            allow_null = all(ref.node.allow_null for ref in self.starstar_arg.entry.cf_references)
-            if allow_null:
-                code.putln("%s = NULL;" % (self.starstar_arg.entry.cname,))
```

Comment:
I think this was intended as an optimisation if the `**kwargs` argument is not used (and thus, the dict not needed) but just written as a catch-all argument to ignore further keyword arguments. Isn't that what this does?

Reply:
If the keyword argument is genuinely not used then the whole block is skipped (see up a bit with line 3882).

This line happens when:
a) no keyword arguments are passed at runtime,
b) there are references to the `kwargs`, but they all have the `allow_null` flag set.

The only way I can see to have a reference with `allow_null` set is to call things like `locals()`, `var()`, etc. In this case it's *safe* for it to be NULL (it means it doesn't appear in the dict of locals) but it really should appear in locals as an empty dict.